src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 export interface Figure {
-
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  sides: number[];
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Length of side should be grater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('These parameters cannot create a triangle');
+    }
 
+    this.shape = 'triangle';
+    this.color = color;
+    this.sides = [a, b, c];
+  }
+
+  getArea(): number {
+    const [a, b, c] = this.sides;
+    const areaT = (a + b + c) / 2;
+
+    const areaTr = Math.sqrt(areaT * (areaT - a) * (areaT - b) * (areaT - c));
+
+    return Math.floor(areaTr * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  color: Color;
 
+  radius: number;
+
+  constructor(color: Color, radius: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again